source/extensions/filters/http/dynamic_modules/abi_impl.cc

```diff
@@ -34,6 +34,19 @@ size_t getHeaderValueImpl(const Http::HeaderMap* map,
   return values.size();
 }
 
+bool headerAsAttribute(const Http::HeaderMap* map, const Envoy::Http::LowerCaseString header,
```

Comment:
let's pass by reference so we can avoid copying 

```suggestion
bool headerAsAttribute(const Http::HeaderMap* map, const Envoy::Http::LowerCaseString& header,
```